zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterProcess.java

```diff
@@ -261,6 +261,12 @@ public int getNumIdleClient() {
     }
   }
 
+  public void setMaxPoolSize(int size) {
+    if (clientPool != null) {
+      //Size + 2 for progress poller , cancel operation
+      clientPool.setMaxTotal(size + 2);
```

Comment:
If default value of pool size is 8, then it makes sense configure the value larger than `maxPoolSize`
But it's little bit weird, because of according to [commons-pool2 api doc](https://commons.apache.org/proper/commons-pool/apidocs/src-html/org/apache/commons/pool2/impl/GenericKeyedObjectPoolConfig.html#line.43), 
DEFAULT_MAX_TOTAL is -1 and no upper limit of the pool size.